sdk/core/azure-core/src/main/java/com/azure/core/http/policy/RetryPolicy.java

```diff
@@ -65,7 +65,8 @@ public RetryPolicy(int maxRetries, Duration delayDuration) {
                     return attemptAsync(context, next, originalHttpRequest, tryCount + 1)
                         .delaySubscription(this.delayDuration);
                 } else {
-                    return Mono.error(err);
+                    return Mono.error(new RuntimeException("Max retries exceeded. Error Details: " + err.getMessage(),
```

Comment:
Do we want to add how many retried were attempted?

Reply:
Yes

Reply:
Done.